src/Toolbar.cc

```diff
@@ -502,12 +502,6 @@ void Toolbar::reconfigure() {
     // we're done with all resizing and stuff now we can request a new
     // area to be reserved on screen
     updateStrut();
-
-#ifdef XINERAMA
```

Comment:
why is this section gone? did valgrind point out this as well?

Reply:
The valgrind error refers to dangling `XineramaHeadMenuItem` pointers. The reason, is that `reloadHeads()` removes these items and create new ones. Calling reloadHeads() seems unnecessary here (its not done for the slits heads menu, btw), and is the easiest way to untangle this.

Reply:
so, how was this triggered in the first place? a click in the xinerama-head-menu lead to the destruction of the menu item which is kind of still in the midst of handling the click?

Reply:
Yes, exactly.

Reply:
```
    if (need_update) {
  …
        menu().removeAll();
        setupMenus(true); // rebuild menu but skip rebuild of placement menu
```
in that case (which seems the valgrind path) "m_xineramaheadmenu->reloadHeads();" would be superflous, otherwise probably not?

Also the only way I can see this causing a problem is if there initially was m_xineramaheadmenu but then something changed to get screen().hasXinerama() to fail and we end up w/ a stale pointer.
So instead have "::setupMenus()" 

```
+   m_xineramaheadmenu = nullptr;
    if (screen().hasXinerama()) {

        m_xineramaheadmenu = new XineramaHeadMenu<Toolbar>(screen().menuTheme(),
``` to make sure we're not running into that condition?